dts: bindings: counter: Change the property names in the overlay

rruuaanng opened the pull request:
Change the property names in the bindings and overlay to use hyphens(-) for separation instead of underscores(_).

kartben reviewed (commented):
same two comments as #83840, both for commit message being "backwards", and need for migration guide entry

kartben reviewed (changes requested):
same two comments as #83840, both for commit message being "backwards", and need for migration guide entry

rruuaanng committed:
dts: bindings: counter:  Change the property names in the overlay

Rename the following properties in binding and overlay:
-- primary_source => primary-source
-- secondary_source => secondary-source
-- filter_count => filter-count
-- filter_period => filter-period

Signed-off-by: James Roy <rruuaanng@outlook.com>

kartben commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -115,6 +115,14 @@ ADC
+Counter
+=======
+
+* Renamed the devicetree propertie ``primary_source`` to ``primary-source``.
+* Renamed the devicetree propertie ``secondary_source`` to ``secondary-source``.
+* Renamed the devicetree propertie ``filter_count`` to ``filter-count``.
+* Renamed the devicetree propertie ``filter_period`` to ``filter-period``.
```
```suggestion
* Renamed the devicetree property ``primary_source`` to ``primary-source``.
* Renamed the devicetree property ``secondary_source`` to ``secondary-source``.
* Renamed the devicetree property ``filter_count`` to ``filter-count``.
* Renamed the devicetree property ``filter_period`` to ``filter-period``.
```

rruuaanng committed:
doc: Add a new migration guide entry to counter

Rename the primary_source, secondary_source,
filter_count and filter_period property to
primary-source,  secondary-source, filter-count
and filter-period in the devicetree and bindings
of the counter subsystem.

Signed-off-by: James Roy <rruuaanng@outlook.com>
